### PR TITLE
MAGECLOUD-3559: [Spike] Check how mutagen.io can improve performance of cloud docker environment

### PR DESCRIPTION
**Pull request by arhiopterecs**
https://magento2.atlassian.net/browse/MAGECLOUD-3559

### Description
Adds the ability to use the Mutagen to synchronize files between the local host and the Docker

### Fixed Issues (if relevant)
1. magento/ece-tools#MAGECLOUD-3559: [Spike] Check how mutagen.io can improve performance of cloud docker environment

### Manual testing scenarios
https://jira.corp.magento.com/browse/MAGECLOUD-124

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Docker/Compose/DeveloperCompose.php [hunk 1]
```diff
@@ -14,16 +14,23 @@
  */
 class DeveloperCompose extends ProductionCompose
 {
+    const SYNC_ENGINE_DOCKER_SYNC = 'docker-sync';
+    const SYNC_ENGINE_MUTAGEN = 'mutagen';
+    const SYNC_ENGINE = 'sync-engine';
```
Comment (YPyltiai) [inline src/Docker/Compose/DeveloperCompose.php:19]: Is this one default sync?

Comment (arhiopterecs) [reply src/Docker/Compose/DeveloperCompose.php:19]: `SYNC_ENGINE = 'sync-engine'` - it is only a key
The default value is `docker-sync`.

File: src/Command/Docker/Build.php [hunk 2]
```diff
@@ -199,6 +220,8 @@ public function execute(InputInterface $input, OutputInterface $output)
             return 1;
         }
 
+        $config->set(DeveloperCompose::SYNC_ENGINE, $syncEngine);
```
Comment (oshmyheliuk) [inline src/Command/Docker/Build.php]: `sync-engine` is already should be set on line 204, why do you need to set it again?

**Review by shiftedreality — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Pull request merged**